__tests__/app.test.js

```diff
@@ -44,5 +43,31 @@ describe("/api", () => {
                 })
             })
         })
+        describe("/articles/:id", () => {
+            test("returns object with correct keys", () => {
+                return request(app)
+                .get("/api/articles/1")
+                .expect(200)
+                .then(({body}) => {
+                    const { article } = body
+                    const ArtKeys = Object.keys(article[0])
+                    expect(Object.keys(article).length).toBeGreaterThan(0)
+                    expect(ArtKeys).toContain("author")
+                    expect(ArtKeys).toContain("title")
+                    expect(ArtKeys).toContain("article_id")
+                    expect(ArtKeys).toContain("body")
+                    expect(ArtKeys).toContain("topic")
+                    expect(ArtKeys).toContain("created_at")
+                    expect(ArtKeys).toContain("votes")
+                    expect(ArtKeys).toContain("article_img_url")
+                    
+                })
+            })
+            test("Returns 404 with invalid ID", () => {
+                return request(app)
+                .get("/api/articles/20000")
+                .expect(404)
+            })
```

Comment:
You should have a .then here to check the message too. A small change to the test description I would say that 20000 is valid, just not found.

You should have another test for an invalid ID however, something like /articles/notanumber. This would be invalid as the data type is incorrect. SQL should throw you an error here